src/main/java/org/killbill/billing/client/KillBillClient.java

```diff
@@ -1913,14 +1946,25 @@ private String getResourceFile(final String uri, final String contentType) throw
 
     private <ReturnType> ReturnType uploadFile(final String fileToUpload, final String uri, final String contentType, final String createdBy, final String reason, final String comment, final Class<ReturnType> followUpClass) throws KillBillClientException {
         Preconditions.checkNotNull(fileToUpload, "fileToUpload cannot be null");
+        final File catalogFile = new File(fileToUpload);
+        Preconditions.checkArgument(catalogFile.exists() && catalogFile.isFile() && catalogFile.canRead(), "file to upload needs to be a valid file");
+        try {
+            final InputStream fileStream = Files.asByteSource(catalogFile).openStream();
```

Comment:
> I updated the code to close the file stream

:+1: 

> As for the InputStream, I figured whoever opens it should close it to avoid confusion.

Yup, makes sense.